### PR TITLE
Fix fitsdiff with a directory argument

### LINKED ISSUE #6760: Comparing a FITS file with one in a subdirectory
I knew that when running fitsdiff to compare a file in the default directory with one of the same name in a subdirectory that it was necessary to explicitly give the name of the file in the subdirectory, but today I typed the following by mistake:

    fitsdiff dms_230_491_refpix.fits today_prev/

and fitsdiff reported "No differences found."  However, I knew that the files did differ, so I ran fitsdiff again:

    fitsdiff dms_230_491_refpix.fits today_prev/dms_230_491_refpix.fits

and fitsdiff showed several differences.

It's probably relevant that there was only one file in `today_prev/`.

### PR DESCRIPTION
**Pull request by agowardhan**
Added unit test for fitsdiff and change to changelog.rst [#6834]

EDIT: Fix #6760

### CODE REVIEW

**Comment by astrofrog** [comment]
@agowardhan - could you edit the title of the pull request to better describe what this PR does?

File: CHANGES.rst [hunk 1]
```diff
@@ -332,6 +332,8 @@ astropy.io.fits
 
 - Fixed potential problems with the compression module [#6732]
 
+- Fixed ``fitsdiff`` script to correctly find the path for .fits files when directories are passed as arguments. [#6834]
```
Comment (pllim) [inline CHANGES.rst]: This needs to be updated to reflect the new PR number (6993).

**Comment by saimn** [comment]
@agowardhan - Your PR contains unrelated changes, it seems something when wrong during the rebase/squash.

**Comment by saimn** [comment]
Closing as the PR stalled with unrelated changes and no progress since, and there is another attempt to fix this issue in #7085. Thank for your efforts anyway @agowardhan !

**Pull request closed without merging**